### PR TITLE
Board scale

### PR DESCRIPTION
**Pull request by erwang01**
Updated Woojin's scaling to scale the entire ARAnchor rather than just the tiles.

Please let me know if it makes sense. Reverting is okay by me as well.

### CODE REVIEW

**Comment by erwang01** [comment]
Also if possible, please test on your side because I wasn't able to test with TileScene.

**Comment by erwang01** [comment]
perhaps i should also consider scaling by sqrt(scale) because right now a scale factor of 0.5 corresponds to 0.25 the original size. (for 3d its 0.125)

**Review by sanathsengupta — COMMENTED** [review]
Seems alright, I can test later — one thing to keep in mind, though, is if the scaling will also work nice with the tile model that we place after board gen.

Also, is there a specific reason you are copying the ground anchor instead of building off of it?

**Comment by erwang01** [comment]
Oh thats because when the minigame plays, we want to disable the board itself (esp once we add all the background and what not).

Currently the board tiles is attached directly onto the AnchorEntity (aka the ground anchor) so disabling all the individual pieces of the board tiles is too much work. Copying the AnchorEntity allows us to disable the board's AnchorEntity in one go.

An alternative to this is have the AnchorEntity have 2 children: Board and Minigame
Then Board can contain all the tiles. / background
and Minigame can contain the minigame stuff
We can then just toggle the Board or the Minigame Entity to enable or disable it.

File: ARrehab/ARrehab/GameBoard.swift [hunk 1]
```diff
@@ -36,8 +36,8 @@ class GameBoard {
     //Minimum extents for a valid game board (in meters)
     //Does not distinguish between x and z directions
 
-    static let EXTENT1 : Float = 2.0 / Tile.SCALE
-    static let EXTENT2 : Float = 3.0 / Tile.SCALE
+    static let EXTENT1 : Float = 2.0 * Tile.SCALE
+    static let EXTENT2 : Float = 3.0 * Tile.SCALE
     
     //Dimension of the board (in tiles)
     static let DIMENSIONS = (3, 5)
```
Comment (erwang01) [inline ARrehab/ARrehab/GameBoard.swift:37]: Sync between DIMENSIONS and EXTENT

File: ARrehab/ARrehab/Tile.swift [hunk 2]
```diff
@@ -14,11 +14,11 @@ class Tile : Entity, HasModel, HasCollision {
     
     //Class attributes
     static let SCALE: Float = 1
-    static let TILE_SIZE = SIMD3<Float>(0.5 / SCALE, 0.01, 0.5 / SCALE)
+    static let TILE_SIZE = SIMD3<Float>(0.5 * SCALE, 0.01 * SCALE, 0.5 * SCALE)
     static let TILE_COLLISION_GROUP = CollisionGroup(rawValue: 1) //Totally arbitrary number
    
     static let defaultTileModel = ModelComponent(mesh: MeshResource.generateBox(size: Tile.TILE_SIZE, cornerRadius: 0.2), materials: [SimpleMaterial()])
-    static let defaultCollisionComp = CollisionComponent(shapes: [ShapeResource.generateBox(width: Tile.TILE_SIZE.x, height: 4.0, depth: Tile.TILE_SIZE.z)], mode: .trigger, filter: CollisionFilter(group: Tile.TILE_COLLISION_GROUP, mask: Player.PLAYER_COLLISION_GROUP))
+    static let defaultCollisionComp = CollisionComponent(shapes: [ShapeResource.generateBox(width: Tile.TILE_SIZE.x, height: 4.0 / SCALE, depth: Tile.TILE_SIZE.z).offsetBy(translation: SIMD3<Float>(0,2,0))], mode: .trigger, filter: CollisionFilter(group: Tile.TILE_COLLISION_GROUP, mask: Player.PLAYER_COLLISION_GROUP))
```
Comment (erwang01) [inline ARrehab/ARrehab/Tile.swift]: keep height at 4 no need to scale up or down.

File: ARrehab/ARrehab/TileGrid.swift [hunk 3]
```diff
@@ -39,6 +39,8 @@ class TileGrid {
         self.gridEntity.transform.translation = surfaceAnchor.center
         
         self.generatePossibleTiles()
+        // Scales the grid appropriately.
+        self.gridEntity.transform.scale = SIMD3<Float>(Tile.SCALE, Tile.SCALE, Tile.SCALE)
```
Comment (erwang01) [inline ARrehab/ARrehab/TileGrid.swift]: dont scale the grid. This is already covered by the internal code.

**Review by erwang01 — COMMENTED** [review]
Discovered that the code already allows for variable tile sizes. mark for changes before merge.

**Comment by erwang01** [comment]
Okay I fixed the scaling to use TILE_SIZE.
One thing to note tho is this branch is currently using Block.usdz and has deleted TileScene.rcproject.

**Pull request merged**